acceptance/tests/environment/variables_refreshed_each_compilation.rb

```diff
@@ -15,62 +15,80 @@
   CONF
   # the module function loading logic is different from inside a single manifest
   #   we exercise both here
-  on(master, "mkdir -p #{fq_tmp_environmentpath}/modules/uptime/{manifests,functions}")
-  create_remote_file(master, "#{fq_tmp_environmentpath}/modules/uptime/manifests/init.pp", <<-FILE)
-    class uptime {
-      notify { 'current uptime':
-        message => uptime::my_system_uptime()
+  on(master, "mkdir -p #{fq_tmp_environmentpath}/modules/custom_time/{manifests,functions,facts.d}")
+  create_remote_file(master, "#{fq_tmp_environmentpath}/modules/custom_time/manifests/init.pp", <<-FILE)
+    class custom_time {
+      $t = custom_time::my_system_time()
+
+      notify { 'custom time':
+        message => "module_${t}_module",
       }
     }
   FILE
-  create_remote_file(master, "#{fq_tmp_environmentpath}/modules/uptime/functions/my_system_uptime.pp", <<-FILE)
-    function uptime::my_system_uptime() {
-      $facts['system_uptime']
+  create_remote_file(master, "#{fq_tmp_environmentpath}/modules/custom_time/functions/my_system_time.pp", <<-FILE)
+    function custom_time::my_system_time() {
+      $facts['custom_time']
     }
   FILE
   create_sitepp(master, tmp_environment, <<-SITE)
     function bar() {
-      $facts['system_uptime']['seconds']
+      $facts['custom_time']
     }
     class foo::bar {
       notify { "local_${bar()}_local": }
     }
     include foo::bar
-    include uptime
+    include custom_time
   SITE
+  create_remote_file(master, "#{fq_tmp_environmentpath}/modules/custom_time/facts.d/custom_time.sh", <<-FILE)
+#!/bin/bash
+
+echo -n "custom_time=$(date +%s%N)"
+  FILE
+  create_remote_file(master, "#{fq_tmp_environmentpath}/modules/custom_time/facts.d/custom_time.ps1", <<-FILE)
+  echo "custom_time=$(get-date -format HHmmssffffff)"
+  FILE
   on(master, "chmod -R 0777 #{fq_tmp_environmentpath}/")
 
-  step "run agent in #{tmp_environment}, ensure it increments the uptime with each run" do
+
+  step "run agent in #{tmp_environment}, ensure it increments the customtime with each run" do
     with_puppet_running_on(master,{}) do
-      uptime = nil
-      module_uptime = nil
-      local_uptime_pattern = 'local_(\d+)_local'
+      local_custom_time_pattern = 'local_(\d+)_local'
+      module_custom_time_pattern = 'module_(\d+)_module'
       agents.each do |agent|
+        # ensure our custom facts have been synced
+        on(agent,
+           puppet("agent -t --server #{master.hostname} --environment #{tmp_environment}"),
+           :accept_all_exit_codes => true)
+
+        local_custom_time1 = module_custom_time1 = nil
+        local_custom_time2 = module_custom_time2 = nil
+
         on(agent, puppet("agent -t --server #{master.hostname} --environment #{tmp_environment}"),
            :accept_all_exit_codes => true) do |result|
           assert_equal(2, result.exit_code, 'wrong exit_code')
```

Comment:
Why don't you set ":acceptable_exit_codes => [ 2 ]" instead of accepting them all and then forcing it to be "2" in the assert_equal check?

Reply:
Because that was the state of the test before I got to it?   (: 
I can change that if you'd prefer.

Reply:
It'd be nice

Reply:
updated